### ISSUE TITLE
path substitution not working from ondeck

### ISSUE THREAD

**Issue by zombieagain** [croneter/PlexKodiConnect#505]
Hi, I started using PKC just a couple of days back. I'm using the stable version with kodi krypton and titan beta.
Yesterday the widgets for ondeck worked ok, but today no video is playing from ondeck. I was able to figure out that the problem is in the path substitution not being executed.

this is the log from the attempt to play from the ondeck widget (I'm pasting the log from the line just after the remote button press):

`08:48:08.324 T:4078797392   DEBUG: Loading settings for /home/joy/shares/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4
08:48:08.331 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers(/home/joy/shares/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4)
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: system rules
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: matches rule: system rules
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: mms/udp
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: lastfm/shout
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: rtmp
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: rtsp
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: streams
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: dvd
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: dvdimage
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: sdp/asf
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: nsv
08:48:08.331 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: radio
08:48:08.332 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: matched 0 rules with players
08:48:08.332 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: adding videodefaultplayer (VideoPlayer)
08:48:08.332 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: for video=1, audio=0
08:48:08.332 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: for video=1, audio=1
08:48:08.332 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: added 1 players
08:48:08.334 T:4078797392   DEBUG: Radio UECP (RDS) Processor - new CDVDRadioRDSData
08:48:08.334 T:4078797392  NOTICE: VideoPlayer: Opening: /home/joy/shares/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4
08:48:08.334 T:4078797392 WARNING: CDVDMessageQueue(player)::Put MSGQ_NOT_INITIALIZED
08:48:08.334 T:4078797392   DEBUG: LinuxRendererGL: Cleaning up GL resources
08:48:08.335 T:2864313088   DEBUG: Thread VideoPlayer start, auto delete: false
08:48:08.335 T:2864313088  NOTICE: Creating InputStream
08:48:08.336 T:2864313088   ERROR: GetDirectory - Error getting /home/joy/shares/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/
08:48:08.336 T:2864313088   ERROR: Previous line repeats 1 times.
08:48:08.336 T:2864313088   DEBUG: CFileCache::Open - opening </home/joy/shares/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4> using cache
08:48:08.337 T:2864313088   ERROR: Open - failed to open source </home/joy/shares/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4>
08:48:08.337 T:2864313088   ERROR: CVideoPlayer::OpenInputStream - error opening [/home/joy/shares/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4]
08:48:08.337 T:2864313088  NOTICE: CVideoPlayer::OnExit()
08:48:08.337 T:2864313088   DEBUG: OnPlayBackStopped: play state was 1, starting 1
08:48:08.337 T:2864313088   DEBUG: Thread VideoPlayer 2864313088 terminating
08:48:08.337 T:4078797392   DEBUG: OnPlayBackStopped: play state was 3, starting 0
08:48:08.337 T:4078797392   ERROR: PlayMedia could not play media: /home/joy/shares/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4
08:48:08.337 T:4078793472   DEBUG: CAnnouncementManager - Announcement: OnStop from xbmc
08:48:08.337 T:4078793472   DEBUG: GOT ANNOUNCEMENT, type: 1, from xbmc, message OnStop
08:48:08.351 T:4078797392  NOTICE: CVideoPlayer::CloseFile()
08:48:08.351 T:4078797392  NOTICE: VideoPlayer: waiting for threads to exit
08:48:08.351 T:4078797392  NOTICE: VideoPlayer: finished waiting
08:48:08.351 T:4078797392   DEBUG: DeleteRenderer - deleting renderer
08:48:08.351 T:4078797392   DEBUG: LinuxRendererGL: Cleaning up GL resources
08:48:08.351 T:4078797392  NOTICE: CVideoPlayer::CloseFile()
08:48:08.351 T:4078797392  NOTICE: VideoPlayer: waiting for threads to exit
08:48:08.351 T:4078797392  NOTICE: VideoPlayer: finished waiting
08:48:08.353 T:4078797392   DEBUG: Radio UECP (RDS) Processor - delete ~CDVDRadioRDSData
08:48:08.354 T:4078797392   DEBUG: LIRC: Update - NEW at 2318543:160 0 KEY_OK_UP linux-input-layer (KEY_OK_UP)
08:48:08.364 T:3783848704   DEBUG: PLEX.kodimonitor: Method: Player.OnStop Data: {u'item': {u'type': u'movie', u'title': u''}, u'end': False}
08:48:08.364 T:3783848704   DEBUG: PLEX.kodimonitor: playback_cleanup called. Active players: set([])
08:48:08.365 T:3783848704   DEBUG: PLEX.kodimonitor: Finished PKC playback cleanup
08:48:08.365 T:3817403136   DEBUG: Skin Helper Widgets --> Kodi_Monitor: sender xbmc - method: Player.OnStop  - data: {"end":false,"item":{"title":"","type":"movie"}}
08:48:08.381 T:3825791744   DEBUG: Skin Helper Service --> Kodi_Monitor: sender xbmc - method: Player.OnStop  - data: {"end":false,"item":{"title":"","type":"movie"}}
08:48:08.416 T:3327886080   DEBUG: $$$ [kodi.callbacks] - Task starting for onPlayBackEnded
08:48:08.478 T:4057985792   DEBUG: PLEX.websocket_client: PMS_Websocket: Received message from PMS server: {u'Setting': [{u'group': u'', u'default': False, u'value': False, u'summary': u'', u'hidden': True, u'label': u'', u'type': u'bool', u'id': u'BackgroundQueueIdlePaused', u'advanced': False}], u'type': u'preference', u'size': 1}
`

and this is the log for the successful play from the library view for the same episode:

`08:55:51.845 T:4078797392   DEBUG: OnPlayMedia /mnt/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4
08:55:51.856 T:4078793472   DEBUG: CAnnouncementManager - Announcement: OnAdd from xbmc
08:55:51.856 T:4078793472   DEBUG: GOT ANNOUNCEMENT, type: 2, from xbmc, message OnAdd
08:55:51.862 T:4078797392   DEBUG: Loading settings for /mnt/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4
08:55:51.870 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers(/mnt/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4)
08:55:51.870 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: system rules
08:55:51.870 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: matches rule: system rules
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: mms/udp
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: lastfm/shout
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: rtmp
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: rtsp
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: streams
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: dvd
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: dvdimage
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: sdp/asf
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: nsv
08:55:51.871 T:4078797392   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: radio
08:55:51.871 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: matched 0 rules with players
08:55:51.871 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: adding videodefaultplayer (VideoPlayer)
08:55:51.871 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: for video=1, audio=0
08:55:51.871 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: for video=1, audio=1
08:55:51.871 T:4078797392   DEBUG: CPlayerCoreFactory::GetPlayers: added 1 players
08:55:51.875 T:4078797392   DEBUG: Radio UECP (RDS) Processor - new CDVDRadioRDSData
08:55:51.875 T:4078797392  NOTICE: VideoPlayer: Opening: /mnt/Storage_Pool/Media/TV Bambini/Plex Versions/Optimized for Mobile/Chuggington/S01E03.mp4
`

Basically, when I start from the library, the path substitution (/home/joy/shares/Storage_Pool --> /mnt) works perfectly, but from the ondeck widget it is not working.

As I said the weird thing is that it was working yesterday, but not this morning.

Thanks!

**Comment by croneter** [comment]
Please share your entire DEBUG log file as explained here: https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug

**Comment by zombieagain** [comment]
Thanks! here's the log:
https://pastebin.com/jQeHre6T
13:01:56.706 --> failed attempt to play from ondeck
13:01:59.838 --> successful play from library

**Comment by zombieagain** [comment]
Hey, I actually fond that, reliably, if I set the "extend" option, than the path substitution happens and the playback works, if I unset it, than it doesn't work anymore.
I have to disable the "extend" option because with it enabled i get less shows in the ondeck...

**Comment by croneter** [comment]
Which extend option?

**Comment by zombieagain** [comment]
Sorry, extend on Deck to all shows in PKC settings

**Comment by croneter** [comment]
Fixed, thanks for the report!

**Comment by croneter** [comment]
Get either the newest stable or beta version of PKC and you should be good
